monai/networks/blocks/crossattention.py

```diff
@@ -145,23 +159,29 @@ def forward(self, x: torch.Tensor, context: Optional[torch.Tensor] = None):
         q = q.view(b, t, self.num_heads, c // self.num_heads).transpose(1, 2)  # (b, nh, t,  hs)
         k = k.view(b, kv_t, self.num_heads, c // self.num_heads).transpose(1, 2)  # (b, nh, kv_t, hs)
         v = v.view(b, kv_t, self.num_heads, c // self.num_heads).transpose(1, 2)  # (b, nh, kv_t, hs)
-        att_mat = torch.einsum("blxd,blyd->blxy", q, k) * self.scale
 
-        # apply relative positional embedding if defined
-        att_mat = self.rel_positional_embedding(x, att_mat, q) if self.rel_positional_embedding is not None else att_mat
+        if self.use_flash_attention:
+            x = torch.nn.functional.scaled_dot_product_attention(q, k, v).contiguous()
+        else:
+
+            att_mat = torch.einsum("blxd,blyd->blxy", q, k) * self.scale
+            # apply relative positional embedding if defined
+            att_mat = (
+                self.rel_positional_embedding(x, att_mat, q) if self.rel_positional_embedding is not None else att_mat
```

Comment:
Looks we can only add `rel_pos_embedding` when not using flash attention. I'm not sure whether it's same with the version in the generative repo, I didn't find the reference.
If that's the case, we may also need add a warning if `self.rel_positional_embedding is not None` and `use_flash_attention=True`.

Reply:
Is it better to have a warning on an error, as we've done with save_attn?

Reply:
error would be better.

Reply:
Hi @KumoLiu and @virginiafdez , when I read this thread:
https://github.com/pytorch/pytorch/issues/96099#issuecomment-1480430583
It seems to me that the relative positional embedding can be integrated with `scaled_dot_product_attention` 's `attn_mask` argument. However, it can be slow as it's not taking the "fast path".

Do you think we can keep this option open for users who wants to use flash_attention and rel_pos_embedding?

Reply:
@mingxin-zheng can you create an issue for that feature please? We've got a number of moving parts already with this PR and later ones building on it, but we can come back to this afterward.

Reply:
Created, closed this one.